core/src/main/java/org/jdbi/v3/core/OnDemandExtensions.java

```diff
@@ -72,6 +75,11 @@
         return extensionType.cast(
                 Proxy.newProxyInstance(
                         extensionType.getClassLoader(),
-                        new Class[]{extensionType}, handler));
+                        new Class<?>[] { extensionType }, handler));
+    }
+
+    public static boolean inOnDemandExtension()
+    {
+        return IN_ON_DEMAND_EXTENSION.get();
```

Comment:
I think we need something more closely scoped to the instance.

Example: if the on-demand instance has a default method that calls a `@CreateSqlObject` that creates a `Transactional` type, then calls transactional methods internally, that should be valid usage.